src/main/java/org/opensearch/flowframework/workflow/WorkflowProcessSorter.java

```diff
@@ -91,7 +74,12 @@ public List<ProcessNode> sortProcessNodes(Workflow workflow) {
                 .map(e -> idToNodeMap.get(e.source()))
                 .collect(Collectors.toList());
 
-            ProcessNode processNode = new ProcessNode(node.id(), step, data, predecessorNodes, executor);
+            TimeValue nodeTimeout = Setting.parseTimeValue(
+                (String) node.inputs().getOrDefault(NODE_TIMEOUT_FIELD, NODE_TIMEOUT_DEFAULT_VALUE),
+                TimeValue.ZERO,
+                String.join(".", node.id(), INPUTS_FIELD, NODE_TIMEOUT_FIELD)
+            );
```

Comment:
Should we create a utility method for the above?

Reply:
Updated with a util with more customized error message.